### PR TITLE
add cordic_v1 for stm32u5

### PR DESCRIPTION
**Pull request by tyler-gilbert**
This MR adds support for the cordic_v1 unit for the stm32u5.

This is the last MR I have for now.

### CODE REVIEW

**Comment by lights0123** [comment]
Are you working on a higher-level CORDIC driver? I was going to take a shot at it if you aren't.

File: stm32-data-gen/src/chips.rs [hunk 1]
```diff
@@ -514,6 +514,7 @@ impl PeriMatcher {
             ("STM32L4.*:GFXMMU:.*", ("gfxmmu", "v1", "GFXMMU")),
             ("STM32U5.*:GFXMMU:.*", ("gfxmmu", "v2", "GFXMMU")),
             ("STM32U5.*:ICACHE:.*", ("icache", "v1", "ICACHE")),
+            ("STM32U5.*:CORDIC:.*", ("cordic", "v1", "CORDIC")),
```
Comment (lights0123) [inline stm32-data-gen/src/chips.rs:517]:
```suggestion
            (".*:CORDIC:.*", ("cordic", "v1", "CORDIC")),
```

All STM chips with CORDIC support (G4, H7, H5, U5) have the same registers, so let's just expose it for all of them.

File: data/registers/cordic_v1.yaml [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+  description: CORDIC co-processor.
+  items:
+  - name: CSR
+    description: Control and status register.
+    byte_offset: 0
+    fieldset: CSR
+  - name: WDATA
+    description: Argument register.
+    byte_offset: 4
+    fieldset: WDATA
```
Comment (lights0123) [inline data/registers/cordic_v1.yaml:11]:
```suggestion
```

File: data/registers/cordic_v1.yaml [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+  - name: RDATA
+    description: Result register.
+    byte_offset: 8
+    fieldset: RDATA
```
Comment (lights0123) [inline data/registers/cordic_v1.yaml:15]:
```suggestion
```

File: data/registers/cordic_v1.yaml [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+    description: Function result.
+    bit_offset: 0
+    bit_size: 32
+fieldset/WDATA:
+  description: Argument register.
+  fields:
+  - name: ARG
+    description: Function input arguments.
+    bit_offset: 0
+    bit_size: 32
```
Comment (lights0123) [inline data/registers/cordic_v1.yaml:83]:
```suggestion
```

**Review by lights0123 — CHANGES_REQUESTED** [review]
These make it rather inconvenient to use as newtypes are generated for simple word types. By removing these, it makes it work like the CRC peripheral so that those registers are just exposed as integers:
image

File: data/registers/cordic_v1.yaml (hunk 3 above)
Comment (tyler-gilbert) [reply data/registers/cordic_v1.yaml:15]: Will fix in the next commit.

File: data/registers/cordic_v1.yaml (hunk 4 above)
Comment (tyler-gilbert) [reply data/registers/cordic_v1.yaml:83]: Got it.

**Comment by tyler-gilbert** [comment]
@lights0123, I will commit the changes you suggested shortly. I haven't worked on a higher level driver and don't plan to at the moment, but from what I read it should be very easy to do that.

**Comment by tyler-gilbert** [comment]
@lights0123, Updated this MR with your suggestions.

**Review by Dirbaio — APPROVED** [review]
thank you!

**Pull request merged**